File: eRestorante/eRestorante.Services/Database/Dish.cs
using System;
using System.Collections.Generic;

namespace eRestorante.Services.Database;

public partial class Dish
{
    public int DishId { get; set; }

    public string DishName { get; set; } = null!;

    public string DishDescription { get; set; } = null!;

    public decimal DishCost { get; set; }

    public byte[]? DishImage { get; set; }

    public int? CategoryId { get; set; }

    public virtual Category? Category { get; set; }

    public virtual ICollection<CommentDish> CommentDishes { get; set; } = new List<CommentDish>();

    public virtual ICollection<OrderDish> OrderDishes { get; set; } = new List<OrderDish>();

    public virtual ICollection<RatingDish> RatingDishes { get; set; } = new List<RatingDish>();
}

File: eRestorante/eRestoranteUI/erestorante_desktop/lib/models/dish.dart
import 'dart:ffi';

import 'package:erestorante_desktop/models/commentDish.dart';
import 'package:erestorante_desktop/models/orderDishes.dart';
import 'package:erestorante_desktop/models/ratingDishes.dart';
import 'package:json_annotation/json_annotation.dart';

part 'dish.g.dart';

@JsonSerializable(explicitToJson: true)
class Dish{
  int? dishID;
  String? dishName;
  String? dishDescription;
  double? dishCost;
  int? categoryId;
  String? dishImage;
  bool? speciality;
  List<CommentDish>? commentDishes;
  List<OrderDishes>? orderDishes;
  List<RatingDishes>? ratingDishes;

  Dish(this.dishID, this.dishName, this.dishDescription, this.dishCost, this.categoryId, this.dishImage, this.speciality, this.commentDishes, this.ratingDishes, this.orderDishes);
  factory Dish.fromJson(Map<String, dynamic> json) => _$DishFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$DishToJson(this);
}